streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -1246,7 +1247,12 @@ public void cleanUp() {
         }
 
         log.debug("Current changelog positions: {}", allChangelogPositions);
-        final Map<TopicPartition, ListOffsetsResultInfo> allEndOffsets = fetchEndOffsetsWithoutTimeout(allPartitions, adminClient);
+        final Map<TopicPartition, ListOffsetsResultInfo> allEndOffsets =
+            fetchEndOffsets(
+                allPartitions,
+                adminClient,
+                getAdminDefaultApiTimeoutMs(config)
```

Comment:
The admin configs is built in ```KafkaStreams``` construction. Could we reuse it?

```scala
        adminClient = clientSupplier.getAdmin(config.getAdminConfigs(ClientUtils.getSharedAdminClientId(clientId)));
```

Reply:
I did it this way since we need to get the admin's default.api.timeout in other places where we only have the `streamsConfig`, so we may as well just pass that as the argument to `getAdminDefaultApiTimeoutMs`

Reply:
We should update the JavaDocs that this method may throw a `TimeoutException` now. What make we wondering if this is a public API change? Was there any discussion on the original KIP about the behavior of `allLocalStorePartitionLags` ?

Reply:
This change might require a KIP... \cc @vvcephei @guozhangwang WDYT?